File: modules/webrtc-core/src/main/java/com/shterneregen/securelan/webrtc/runtime/RtcRuntimeStatus.java
package com.shterneregen.securelan.webrtc.runtime;

public record RtcRuntimeStatus(String providerName, boolean available, String message) {
    public static RtcRuntimeStatus unavailable(String message) {
        return new RtcRuntimeStatus("unconfigured", false, message);
    }
}

File: modules/webrtc-core/src/main/kotlin/com/shterneregen/securelan/webrtc/runtime/RtcRuntimeStatus.kt
package com.shterneregen.securelan.webrtc.runtime

@JvmRecord
data class RtcRuntimeStatus(
    val providerName: String?,
    val available: Boolean,
    val message: String?,
) {
    companion object {
        @JvmStatic
        fun unavailable(message: String?): RtcRuntimeStatus = RtcRuntimeStatus("unconfigured", false, message)
    }
}
